### PR TITLE
Shutdown analysis queue

### PR DESCRIPTION
**Pull request by vdimir**
Hi!

In package `bleve` function `init` runs several goroutines for procession analysis queue.
In some cases we need to shutdown this goroutines. For example, it triggers goleak tool.

```
package main

import (
	"testing"
	"github.com/blevesearch/bleve"
	"go.uber.org/goleak"
)

func main() {
	var idx bleve.Index
	_ = idx
}

func TestMain(m *testing.M) {
	goleak.VerifyTestMain(m)
}
```

Causes:
```
goleak: Errors on successful test run: found unexpected goroutines:
...
```

Also, when `SetAnalysisQueueSize` is called analysis queue size not set to specified size, but increased on this number because exited workers isn't closed.

This PR adds `Shutdown` function that closes analysis queue from user's code and closes queue when new size is set.

### CODE REVIEW

**Comment by vdimir** [comment]
Also there are other question. Should we always run analysis queue by default? Maybe it is better not run in it `init`, but allow user to start process? But this PR provide conservative solution that leaves all functionality as is.

**Review by paskal — APPROVED** [review]

**Comment by mschoch** [comment]
I think there is already consensus that the design is not a good one.  But fixing it properly will be a breaking change, so that will have to wait for 2.0.

I would be OK with this as an interim solution, thoughts @abhinavdangeti @sreekanth-cb ?

**Comment by vdimir** [comment]
Any updates on this?

**Review by abhinavdangeti — APPROVED** [review]
Ok with this. This can be merged after one other approval from the team.

**Review by mschoch — APPROVED** [review]

**Pull request merged**
